test/extended/builds/pipeline.go

```diff
@@ -93,6 +93,41 @@ var _ = g.Describe("[Feature:Builds][Slow] openshift pipeline build", func() {
 			if os.Getenv(jenkins.EnableJenkinsMemoryStats) != "" {
 				ticker.Stop()
 			}
+
+			// per k8s e2e volume_util.go:VolumeTestCleanup, nuke any client pods
+			// before nfs server to assist with umount issues; as such, need to clean
+			// up prior to the AfterEach processing, to guaranteed deletion order
+			client := oc.AsAdmin().KubeFramework().ClientSet
+			if jenkinsTemplatePath == jenkinsPersistentTemplatePath {
+				g.By("deleting PVC")
+				pvclist, err := client.CoreV1().PersistentVolumeClaims(oc.Namespace()).List(metav1.ListOptions{})
+				if err != nil {
+					e2e.Logf("pvc list error %#v\n", err)
+				}
+				for _, pvc := range pvclist.Items {
+					g.By("found pvc " + pvc.Name)
+					if strings.HasPrefix(pvc.Name, "jenkins") {
+						err = client.CoreV1().PersistentVolumeClaims(oc.Namespace()).Delete(pvc.Name, nil)
```

Comment:
these PVCs are presumably claimed by running pods, that doesn't present a problem when deleting them?

(would have thought you'd delete jenkins, then the PVCs, then the PVs, then the NFS server)

Reply:
Yeah I did it in this order consciously based on my interpretation of some of the comments around existing issues with unmounting nfs volumes that I saw in the k8s e2e code (the files I noted in the comments in the `cleanup()` function.

Once everything else is stabilized, I could go back and experiment with the order some more I suppose.

But if the order is not *right* the unmounts and pods terminations will hang.

Reply:
i guess as long as this order appears to work it's fine :)
just slightly surprised.  not founded on any expertise of how it works though.

Reply:
ok I tried the delete app, delete pvcs, delete pvs, delete nfs pod, and that worked as well;
as long as you don't delete the nfs pod before the app pod, it is OK

I"ll update the mongo/postgres/jenkins/mysql to follow the more intuitive order